consensus/qsc/mod.go

```diff
@@ -67,38 +69,59 @@ func (c *Consensus) GetChain(id []byte) (consensus.Chain, error) {
 // primitives to send proposals to a network of nodes.
 func (c *Consensus) Listen(val consensus.Validator) (consensus.Actor, error) {
 	go func() {
-		// TODO: how to stop the loop ?
 		for {
-			err := c.executeRound(val)
+			var proposal consensus.Proposal
+			select {
+			case <-c.closing:
+				fabric.Logger.Trace().Msg("closing")
+				return
+			case proposal = <-c.ch:
+			default:
```

Comment:
That seems weird: if there isn't any proposals in `c.ch`, then this loops forever at full speed calling executeRound() with a nil proposal?

Reply:
Well, either you have a proposal or none but you need to participate to the consensus anyway. It odes not loop at full speed because you have a time step execution every time.